spec-tests/src/test/java/org/candlepin/spec/owners/OwnerResourcePoolFilterSpecTest.java

```diff
@@ -0,0 +1,322 @@
+        providedProducts.add(createProductByOwner(adminClient, owner, Products.random()));
+        providedProducts.add(createProductByOwner(adminClient, owner, Products.random()));
+        ProductDTO product = createProductByOwner(
+            adminClient, owner, Products.random().providedProducts(providedProducts));
+        adminClient.owners().createPool(owner.getKey(), Pools.random(product));
+
+        List<PoolDTO> pools = adminClient.owners().listOwnerPools(owner.getKey(), consumer.getUuid(), null,
+            null, null, null, null, List.of(providedName), null,
+            null, null, null, null, null, null, null, null);
+        assertEquals(1, pools.size());
+        assertEquals(owner.getKey(), pools.get(0).getOwner().getKey());
+        assertEquals(product.getId(), pools.get(0).getProductId());
+        assertEquals(3, pools.get(0).getProvidedProducts().size());
+    }
+
+    @Test
+    @DisplayName("should allow user to list standard pool by subscription id")
+    public void shouldAllowUserToListStandardPoolBySubscriptionId() throws ApiException {
+        ApiClient adminClient = ApiClients.admin();
+        HostedTestApi upstreamClient = adminClient.hosted();
+        OwnerDTO owner = createOwner(adminClient);
+        ApiClient userClient = createUserClient(adminClient, owner);
+        ProductDTO product = createProductByOwner(adminClient, owner, Products.random());
+        PoolDTO pool;
+        // entitle owner for the virtual and monitoring products
+        if (CandlepinMode.isHosted()) {
+            createUpstreamProducts(upstreamClient, product);
+            SubscriptionDTO sub1 = Subscriptions.random(owner, product);
+            createSubscriptions(adminClient, upstreamClient, owner, sub1);
+            pool = adminClient.owners().listOwnerPoolsWithSubscription(
+                owner.getKey(), sub1.getId(), true).get(0);
+        }
+        else {
+            pool = adminClient.owners().createPool(owner.getKey(), Pools.random(product));
+        }
+
+        List<PoolDTO> pools = userClient.owners().listOwnerPools(owner.getKey(), null, null,
+            null, pool.getSubscriptionId(), null, null, null, null,
+            null, null, null, null, null, null, null, null);
+        assertEquals(1, pools.size());
+    }
+
+    @Test
+    @DisplayName("should allow user to list bonus pool also by subscription id")
+    public void shouldAllowUserToListBonusPoolAlsoBySubscriptionId() throws ApiException {
+        ApiClient adminClient = ApiClients.admin();
+        HostedTestApi upstreamClient = adminClient.hosted();
+        OwnerDTO owner = createOwner(adminClient);
+        ApiClient userClient = createUserClient(adminClient, owner);
+        ProductDTO product = createProduct(adminClient, owner,
+            ProductAttributes.VirtLimit.withValue("unlimited"));
+        PoolDTO pool;
+
+        if (CandlepinMode.isHosted()) {
```

Comment:
Unnecessary hosted check. Just create the pools as you need them.

Reply:
done